applications/IgaApplication/custom_elements/laplacian_IGA_element.cpp

```diff
@@ -0,0 +1,231 @@
+    Vector temp(number_of_points);
+
+    const double heat_flux = this->GetValue(r_volume_source_var);
+    const double conductivity = this->GetProperties().GetValue(r_diffusivity_var);
+
+    for(IndexType i_point = 0; i_point < integration_points.size(); ++i_point)
+    {
+        noalias(DN_DX) = DN_De[i_point] ; //prod(DN_De[i_point],InvJ0);
+
+        auto N = row(N_gausspoint,i_point);
+        const double IntToReferenceWeight = integration_points[i_point].Weight(); // * std::abs(DetJ0);
+
+        noalias(rLeftHandSideMatrix) += IntToReferenceWeight * conductivity * prod(DN_DX, trans(DN_DX));
+        noalias(rRightHandSideVector) += IntToReferenceWeight * heat_flux * N;
+    }
+
+
+    // RHS = ExtForces - K*temp;
+    for (IndexType i = 0; i < number_of_points; i++) {
+        temp[i] = r_geometry[i].GetSolutionStepValue(r_unknown_var);
+    }
+
+    // RHS -= K*temp
+    noalias(rRightHandSideVector) -= prod(rLeftHandSideMatrix,temp);
+
+    KRATOS_CATCH("")
+}
+
+
+// From classical Laplacian
+void LaplacianIGAElement::CalculateLeftHandSide(MatrixType& rLeftHandSideMatrix, const ProcessInfo& rCurrentProcessInfo)
+{   
+    VectorType temp(0);
+    CalculateLocalSystem(rLeftHandSideMatrix, temp, rCurrentProcessInfo);
+}
+
+
+// From classical Laplacian
+void LaplacianIGAElement::CalculateRightHandSide(VectorType& rRightHandSideVector, const ProcessInfo& rCurrentProcessInfo)
+{
+    MatrixType temp(0,0);
+    CalculateLocalSystem(temp, rRightHandSideVector, rCurrentProcessInfo);
+}
+
+
+
+
+
+void LaplacianIGAElement::EquationIdVector(EquationIdVectorType& rResult, const ProcessInfo& rCurrentProcessInfo) const
+{
+    const ProcessInfo& r_process_info = rCurrentProcessInfo;
+    ConvectionDiffusionSettings::Pointer p_settings = r_process_info[CONVECTION_DIFFUSION_SETTINGS];
+    auto& r_settings = *p_settings;
+
+    const Variable<double>& r_unknown_var = r_settings.GetUnknownVariable();
+
+    unsigned int number_of_nodes = GetGeometry().PointsNumber();
+    if(rResult.size() != number_of_nodes)
+        rResult.resize(number_of_nodes);
+    for (unsigned int i=0; i<number_of_nodes; i++)
+    {
+        rResult[i] = GetGeometry()[i].GetDof(r_unknown_var).EquationId();
+    }
+}
+
+//************************************************************************************
+//************************************************************************************
+void LaplacianIGAElement::GetDofList(DofsVectorType& ElementalDofList,const ProcessInfo& rCurrentProcessInfo) const
+{
+    const ProcessInfo& r_process_info = rCurrentProcessInfo;
+    ConvectionDiffusionSettings::Pointer p_settings = r_process_info[CONVECTION_DIFFUSION_SETTINGS];
+    auto& r_settings = *p_settings;
+
+    const Variable<double>& r_unknown_var = r_settings.GetUnknownVariable();
+
+    unsigned int number_of_nodes = GetGeometry().PointsNumber();
+    if(ElementalDofList.size() != number_of_nodes)
+        ElementalDofList.resize(number_of_nodes);
```

Comment:
```suggestion
    if(ElementalDofList.size() != number_of_nodes) {
        ElementalDofList.resize(number_of_nodes);
    }
```
Either you use the braces for one line if/for or not.